app/data/schemas.server.ts

```diff
@@ -142,3 +142,436 @@ export const BlessingSchema = z
+      displayIndex: z.number(),
+    })
+  ),
+  refreshAt: z.string(),
+  _links: z.unknown(),
+})
+export const LatenciesSchema = z.object({
+  regions: z.array(
+    z.object({
+      region: z.string(),
+      httpLatencyUrl: z.string(),
+    })
+  ),
+})
+
+export const AccountSummarySchema = z.object({
+  summary: z.object({
+    sub: z.string(),
+    name: z.string(),
+    characters: z.array(
+      z.object({
+        id: z.string(),
+        name: z.string(),
+        inventory: z.record(z.string()),
+        breed: z.string(),
+        archetype: z.string(),
+        gender: z.string(),
+        lore: z.unknown(),
+        selectedVoice: z.string().optional(),
+        abilities: z.unknown(),
+        career: z.object({
+          specialization: z.string(),
+          talents: z.array(z.string()).optional(),
+        }),
+        narrative: z.unknown(),
+        personal: z.unknown(),
+        prison_number: z.string(),
+      })
+    ),
+    data: z.array(z.unknown()),
+    currencies: z.array(z.unknown()),
+    statistics: z.array(z.unknown()),
+    consumables: z.array(z.unknown()),
+    activeEffects: z.array(z.unknown()),
+    progression: z.object({
+      type: z.literal("account"),
+      id: z.string(),
+      currentLevel: z.number(),
+      currentXp: z.number(),
+      currentXpInLevel: z.number(),
+      neededXpForNextLevel: z.number(),
+      unclaimedRewards: z.array(z.unknown()),
+      eligibleForLevel: z.boolean(),
+    }),
+  }),
+})
+export const AccountGearSchema = z.object({
+  gearList: z.record(
+    z.object({
+      slots: z.array(z.string()),
+      masterDataInstance: z.object({
+        id: z.string(),
+        overrides: z
+          .object({
+            ver: z.number().optional(),
+            traits: z
+              .array(
+                z.object({
+                  value: z.number().optional(),
+                  id: z.string(),
+                  rarity: z.number(),
+                })
+              )
+              .optional(),
+            perks: z
+              .array(
+                z.object({
+                  value: z.number().optional(),
+                  id: z.string(),
+                  rarity: z.number(),
+                })
+              )
+              .optional(),
+            characterLevel: z.number().optional(),
+            baseItemLevel: z.number().optional(),
+            baseStats: z
+              .array(z.object({ name: z.string(), value: z.number() }))
+              .optional(),
+            rarity: z.number().optional(),
+            itemLevel: z.number().optional(),
+          })
+          .optional(),
+      }),
+      characterId: z.string().optional(),
+    })
+  ),
+})
+
+export const JoinSchema = z.object({
+  queuePosition: z.number(),
+  queueTicket: z.string(),
+  retrySuggestion: z.number(),
+})
+
+
+export const CharactersSchema = z.object({
+  _links: z.object({ self: z.object({ href: z.string() }) }),
+  characters: z.array(
+    z.union([
```

Comment:
Kinda hard to tell with such a big object, what the difference between the character types in this union?

Reply:
I was auto converting the json without actually checking the format